### PR TITLE
Hello World web application using Spring Boot and Gradle.

### PR DESCRIPTION
**Pull request by TerJkee**
Also were added:
-  springfox Swagger2
-  Lombok 
- .gitignore file.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,15 @@
+.gradle
+/build/
```
Comment (PixarV) [inline .gitignore]: Please, delete the first slash: only absolute paths starts with '/'.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+gradle-app.setting
+
+# Avoid ignoring Gradle wrapper jar file (.jar files are usually ignored)
+!gradle-wrapper.jar
+
+# Cache of project
+.gradletasknamecache
+
+#TestSystem project specific:
+/.idea/
```
Comment (PixarV) [inline .gitignore]: Delete the first '/' x2

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+/TestSystem.iml
```
Comment (PixarV) [inline .gitignore]: Delete the first '/' x3 + all iml files can be added by pattern `*.iml`

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        mavenCentral()
+    }
+    dependencies {
+        classpath("org.springframework.boot:spring-boot-gradle-plugin:2.1.7.RELEASE")
+    }
+}
+
+plugins {
+    id 'java'
+    id 'war'
```
Comment (PixarV) [inline build.gradle]: Why do we need this plugin? Spring Boot packs into jars

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,33 @@
+buildscript {
```
Comment (PixarV) [inline build.gradle]: As I remember we don't need a `buildscript` block here. All our plugins are imported without it.

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+group 'org.neorg'
+version '1.0-SNAPSHOT'
+
+repositories {
+    mavenCentral()
+}
+
+sourceCompatibility = 11
```
Comment (PixarV) [inline build.gradle]: Gradle has a constant for this. `JavaVersion.VERSION_11`

File: build.gradle [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        classpath("org.springframework.boot:spring-boot-gradle-plugin:2.1.7.RELEASE")
+    }
+}
+
+plugins {
+    id 'java'
+    id 'war'
+    id 'idea'
+    id 'org.springframework.boot' version '2.1.7.RELEASE'
+    id "io.spring.dependency-management" version "1.0.8.RELEASE"
```
Comment (PixarV) [inline build.gradle]: Let's choose between single and double quotes ) I'm for single

File: src/main/java/testSystem/hello/HelloWorldController.java [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class HelloWorldController {
+
+    @Autowired
+    HelloWorldService helloWorldService;
```
Comment (rootrock13) [inline src/main/java/testSystem/hello/HelloWorldController.java]: where is `private`?
why don't you use autowired via constructor?
why don't you use `lombok` with `@NoArgsConstructor, @RequiredArgsConstructor, @AllArgsConstructor` before the class definition?

File: src/main/java/testSystem/hello/HelloWorldService.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package testSystem.hello;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class HelloWorldService {
+    private static final String HELLO_WORLD_STRING = "Hello, World!";
```
Comment (rootrock13) [inline src/main/java/testSystem/hello/HelloWorldService.java]: redundant code _STRING in the name

File: .gitignore (hunk 1 above)
Comment (PixarV) [reply .gitignore]: You can use `*/build/` also

File: src/main/java/testSystem/hello/HelloWorldService.java [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+
+    public String makeHelloString() {
```
Comment (rootrock13) [inline src/main/java/testSystem/hello/HelloWorldService.java]: same thing + why `make` (`getHelloMessage` would be more suitable)

File: build.gradle [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    mavenCentral()
+}
+
+sourceCompatibility = 11
+
+dependencies {
+    compile group: 'org.springframework.boot', name: 'spring-boot-starter-web', version: '2.1.7.RELEASE'
+    compile group: 'io.springfox', name: 'springfox-swagger2', version: '2.9.2'
+    compile group: 'io.springfox', name: 'springfox-swagger-ui', version: '2.9.2'
+    providedCompile group: 'org.projectlombok', name: 'lombok', version: '1.18.8'
```
Comment (PixarV) [inline build.gradle]: Lombok also requires an additional dependency: `annotationProcessor 'org.projectlombok:lombok:1.18.8'`

File: src/main/java/testSystem/App.java [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+@SpringBootApplication
+@EnableSwagger2
+public class App {
+
+    public static void main(String[] args) {
+        SpringApplication.run(App.class, args);
+    }
+
+    @Bean
+    public Docket api() {
```
Comment (PixarV) [inline src/main/java/testSystem/App.java]: All `@Bean` classes should be placed in package `configurations` via classes like `SwaggerConfiguration.java`

File: src/main/java/testSystem/hello/HelloWorldController.java [hunk 13]
```diff
@@ -0,0 +1,17 @@
+package testSystem.hello;
```
Comment (PixarV) [inline src/main/java/testSystem/hello/HelloWorldController.java]: Let's define the proper structure of packages. Controller -> `controllers`

File: src/main/java/testSystem/App.java [hunk 14]
```diff
@@ -0,0 +1,15 @@
+package testSystem;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import springfox.documentation.swagger2.annotations.EnableSwagger2;
+
+
+@SpringBootApplication
+@EnableSwagger2
```
Comment (PixarV) [inline src/main/java/testSystem/App.java]: You can move this annotation on the SwaggerConfig class. And annotations should be placed like a pyramid:
```java
@EnableSwagger2
@SpringBootApplication
public class App { ...
```

File: src/main/java/testSystem/App.java [hunk 15]
```diff
@@ -0,0 +1,15 @@
+package testSystem;
```
Comment (PixarV) [inline src/main/java/testSystem/App.java]: According to google java style guide: 

> Package names are all lowercase, with consecutive words simply concatenated together (no underscores). For example, com.example.deepspace, not com.example.deepSpace or com.example.deep_space

Rename it, please )

File: build.gradle [hunk 16]
```diff
@@ -0,0 +1,27 @@
+plugins {
+    id 'java'
+    id 'idea'
+    id 'org.springframework.boot' version '2.1.7.RELEASE'
+    id 'io.spring.dependency-management' version '1.0.8.RELEASE'
+}
+
+group 'org.neorg'
```
Comment (shaart) [inline build.gradle]: Is this an appropriate group for the project?

File: build.gradle [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+version '1.0-SNAPSHOT'
```
Comment (shaart) [inline build.gradle]: The very first version of a project is recommended to be set like "0.0.1".
Where:
- "0.X.X" - 0 is Major version. It's incremented when you release new functionality and fix public API.
- "X.0.X" - 0 is Minor version.
- "X.X.1" - 1 is Patch version.
P.S. There could be an additional version number like "Build version" and "Compilation version" - 0.0.1.5.133.
If you use intermediate version - could be used word "SNAPSHOT".

More info here: https://semver.org/

File: build.gradle (hunk 16 above)
Comment (PixarV) [reply build.gradle]: It's a very good question. Maybe `com.rt` )

Comment (shaart) [reply build.gradle]: I guess you should use the name of your project or maybe a potential web address of the system.
Something like `test.system`, `com.test.system`, `com.ts`, `com.tsys` or `org.*`.

**Review by Elanlum — APPROVED** [review]

**Review by PixarV — APPROVED** [review]
Good job :)

**Pull request merged**